src/Typesense/ApiCall.js

```diff
@@ -3,6 +3,17 @@ import axios from 'axios'
 
 const APIKEYHEADERNAME = 'X-TYPESENSE-API-KEY'
 
+const buildLocationUrl = function(host, port) {
+  // if the host is a non-root URL, such as example.com/typesense
+  // then the port number cannot be appended, as example.com/typesense:8108
+  // instead it must be example.com:8108/typesense
```

Comment:
@dpastoor Thanks for the PR! I've a couple of concerns about this approach though:

1. If the `host` property is allowed to be given a path like `example.com/typesense`, then it's technically not a `host` anymore since a host should refer to a IP or domain. 
2. Passing `host = example.com/typesense` and `port = 8108`in the configuration would be a bit confusing as well - if one looks just at just the host part (`example.com/typesense`) , it looks as if the port is omitted and so the service is using the default HTTP port of 80.

Here's an alternative: would it help if we introduced another parameter called `path`? Then the configuration would look like this:

```
var typesense = new Typesense.Client({
    'masterNode': {
      'host': 'example.com',
      'port': '80',
      'path': '/typesense',
      'protocol': 'http',
      'apiKey': 'abcd'
    }
    'timeoutSeconds': 10
  })
```

This makes everything explicit. If a `path` property is provided, we will just append that path when constructing the URL:

```
`${this._configuration.masterNode.protocol}://${this._configuration.masterNode.host}:${this._configuration.masterNode.port}${this._configuration.masterNode.path}${endpoint}`
```

If path is empty, then there is no harm and will be equivalent to how it is today.

Reply:
I'd be completely satisified with a path argument. I just need _some_ way of doing it!